Joystream-Role-Bot/src/controls/control.ts

```diff
@@ -150,13 +155,23 @@ export const setMemberRole = async (client: Client): Promise<void> => {
 
       const roleUpdatePromise =
         memberRole.status.__typename === "WorkerStatusActive"
-          ? discordMember.roles.add(role)
-          : discordMember.roles.remove(role);
+          ? await discordMember.roles.add(role)
+          : await discordMember.roles.remove(role);
+
       await roleUpdatePromise;
+
+      const daoRole = await guild.roles.fetch(RoleAddress.DAO);
```

Comment:
No need to fetch this role for every member separately, you can fetch it once above `qnMembers.map`

Reply:
This is how you get a role in a Discord channel.
But qnmember.map gets its role from QN.
So they are different.

Reply:
I understand that but I said something else. Currently you are fetching the role from Discord inside a loop. There's no need to do that, you should do it just once

Reply:
okay. I understand. I will finished that on the second PR